src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeoAuthProvider.scala

```diff
@@ -24,26 +28,26 @@ object NotebookClusterActions {
 
 }
 
-abstract class LeoAuthProvider(authConfig: Config) {
+abstract class LeoAuthProvider(authConfig: Config, serviceAccountProvider: ServiceAccountProvider) {
   /**
-    * @param userInfo The user in question
+    * @param userEmail The user in question
     * @param action The project-level action (above) the user is requesting
     * @param googleProject The Google project to check in
     * @return If the given user has permissions in this project to perform the specified action.
     */
-  def hasProjectPermission(userInfo: UserInfo, action: ProjectActions.ProjectAction, googleProject: String): Future[Boolean]
+  def hasProjectPermission(userEmail: WorkbenchEmail, action: ProjectActions.ProjectAction, googleProject: String)(implicit executionContext: ExecutionContext): Future[Boolean]
 
   /**
     * Leo calls this method to verify if the user has permission to perform the given action on a specific notebook cluster.
     * It may call this method passing in a cluster that doesn't exist. Return Future.successful(false) if so.
     *
-    * @param userInfo      The user in question
+    * @param userEmail      The user in question
     * @param action        The cluster-level action (above) the user is requesting
     * @param googleProject The Google project the cluster was created in
     * @param clusterName   The user-provided name of the Dataproc cluster
     * @return If the userEmail has permission on this individual notebook cluster to perform this action
     */
-  def hasNotebookClusterPermission(userInfo: UserInfo, action: NotebookClusterActions.NotebookClusterAction, googleProject: String, clusterName: String): Future[Boolean]
+  def hasNotebookClusterPermission(userEmail: WorkbenchEmail, action: NotebookClusterActions.NotebookClusterAction, googleProject: String, clusterName: String)(implicit executionContext: ExecutionContext): Future[Boolean]
```

Comment:
Are these fine as WorkbenchEmail or should it be String?

Reply:
We use `WorkbenchEmail` (and other model classes) in other provider classes. I think it's fine -- the Leo project has a dependency on workbench-libs which should extend to custom provider implementations.

Reply:
Oh - we're also using String for googleProject and clusterName - if we're using model classes for userEmail, then those should use model classes too right?

Reply:
Yeah I think we need to make a decision whether to use only model classes or only Strings. :)